drivers/platform/aducm3029/gpio.c

```diff
@@ -205,6 +205,28 @@ int32_t gpio_get_direction(struct gpio_desc *desc, uint8_t *direction)
 	return SUCCESS;
 }
 
+/**
+ * @brief Place GPIO in tristate mode.
+ * @param desc - The GPIO descriptor.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t gpio_tristate(struct gpio_desc *desc)
```

Comment:
I would add another output gpio state:
GPIO_HIGH_Z 
And use gpio_set_value(..., GPIO_HIGH_Z).

This approach would be backwards compatible and you could get rid of this function.

Reply:
The thing to watch out here is to not change functionality for any of the current projects. As it currently stands if the second parameter of `gpio_set_value` is anything but zero, the GPIO will be set to high state. I will be introduceing a new case, where if the second parameter is 2 the GPIO will be high impedance. I will leave the rest of the cases to bring it to high state.